tests/dags/common/loader/test_reporting.py

```diff
@@ -255,9 +257,55 @@ def test_clean_time_duration(seconds, expected):
     assert actual == expected
 
 
+@pytest.mark.parametrize(
+    "duration, record_counts_by_media_type, expected",
+    [
+        (
+            None,
+            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
+            False,
+        ),
+        (None, {}, True),
+        (
+            "less than 1 sec",
+            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
+            False,
+        ),
+        ("less than 1 sec", {}, True),
+        (
+            "inf",
+            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
+            False,
+        ),
+        ("inf", {}, True),
+        (
+            "10 secs",
+            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
+            False,
+        ),
+        ("10 secs", {}, False),
+        (
+            "16 weeks, 3 days, 17 hours, 46 mins, 40 secs",
+            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
+            False,
+        ),
+        ("16 weeks, 3 days, 17 hours, 46 mins, 40 secs", {}, False),
```

Comment:
Nit: I might suggest reordering these tests just so that they're grouped by expectation rather than by duration + record counts.
```suggestion
        # Should skip
        (None, {}, True),
        ("less than 1 sec", {}, True),
        ("inf", {}, True),
        # Should report metrics
        (
            None,
            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
            False,
        ),
        (
            "less than 1 sec",
            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
            False,
        ),
        (
            "inf",
            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
            False,
        ),
        (
            "10 secs",
            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
            False,
        ),
        ("10 secs", {}, False),
        (
            "16 weeks, 3 days, 17 hours, 46 mins, 40 secs",
            {"image": RecordMetrics(1, 2, 3, 4), "audio": RecordMetrics(1, 2, 0, 0)},
            False,
        ),
        ("16 weeks, 3 days, 17 hours, 46 mins, 40 secs", {}, False),
```